test/libsolidity/smtCheckerTests/array_members/push_function_return_value.sol

```diff
@@ -0,0 +1,20 @@
+contract C {
+	bytes x;
+
+	function getX() internal view returns (bytes storage) {
+		return x;
+	}
+
+	function s() public {
+		require(x.length == 0);
+		getX().push("a");
+		assert(x.length == 1); // should hold but knowledge is erased due to pushing a reference
```

Comment:
The reset is due to it being returned from a function rather than due to the push, right?
```suggestion
		assert(x.length == 1); // should hold but knowledge is erased due to returning a reference
```

Reply:
Not really, the resetting is due to the push, since we don't know exactly what it is. If we don't assign/push the returned reference then nothing needs to be reset.

Reply:
Ok, thought that the assign referred to assigning the returns inside the function. Now it makes more sense.